src/test/java/com/google/collegeplanner/DeletePlanServletTest.java

```diff
@@ -0,0 +1,133 @@
+  DatastoreService datastore;
+  DeletePlanServlet servlet;
+  StringWriter stringWriter;
+  PrintWriter writer;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void before() throws Exception {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+    when(response.getWriter()).thenReturn(writer);
+    servlet = new DeletePlanServlet();
+    helper.setUp();
+  }
+
+  @After
+  public void after() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void servletDeletesDatastoreObjects() throws Exception {
+    // Put 2 entities in datastore.
+    Entity planEntity1 = new Entity("Plan");
+    datastore.put(planEntity1);
+
+    Entity planEntity2 = new Entity("Plan");
+    datastore.put(planEntity2);
+
+    // Delete an element and check that there is only 1 element in datastore.
+    when(request.getParameter("id")).thenReturn("1");
+    servlet.doPost(request, response);
+
+    verify(response, never()).setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    verify(response, never()).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    Assert.assertEquals(1, datastore.prepare(new Query("Plan")).countEntities());
+  }
+
+  @Test
+  public void servletDoesntDeleteInvalidIds() throws Exception {
+    // Put 2 entities in datastore.
+    Entity planEntity1 = new Entity("Plan");
+    datastore.put(planEntity1);
+
+    Entity planEntity2 = new Entity("Plan");
+    datastore.put(planEntity2);
+
+    // Delete an element and check that there are still 2 elements in datastore.
+    when(request.getParameter("id")).thenReturn("3");
+    servlet.doPost(request, response);
+
+    verify(response, never()).setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    verify(response, never()).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    Assert.assertEquals(2, datastore.prepare(new Query("Plan")).countEntities());
+  }
+
+  @Test
+  public void servletReturnsErrorWhenNullParam() throws Exception {
+    // Put 2 entities in datastore.
+    Entity planEntity1 = new Entity("Plan");
+    datastore.put(planEntity1);
+
+    Entity planEntity2 = new Entity("Plan");
+    datastore.put(planEntity2);
+
+    // Delete an element and check that there are still 2 elements in datastore.
+    when(request.getParameter("id")).thenReturn(null);
+    servlet.doPost(request, response);
+
+    verify(response, times(1)).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    verify(response, never()).setStatus(not(eq(HttpServletResponse.SC_BAD_REQUEST)));
+    String expectedJson =
+        "{\"message\":\"Invalid id for datastore deletion.\",\"status\":\"error\"}";
+    JSONAssert.assertEquals(expectedJson, stringWriter.toString(), JSONCompareMode.STRICT);
+  }
```

Comment:
Maybe I missed it but will the delete button be shown when there are no plans?  Because if i's shown you'll need a test for testing out the delete button when there are no saved plans

Reply:
The delete buttons are only created along with the cards for each schedule